### ISSUE TITLE
Reachability: Consider local variables as well?

### ISSUE THREAD

**Issue by RolandKluge** [eMoflon/emoflon-tool#210]
@gervarro Should we consider the local variables of a pattern as well when analyzing the reachability of a pattern?

Currently, target adornment/BDD variables/.. are calculated for the symbolic parameters of a pattern only.

This means that operations that work on local variables are completely ignored during the reachability analysis.
In my opinion, this may lead to false negative results: The reachability analysis may claim that a pattern cannot be matched for a given adornment even though a search plan exists.
In the SoSyM paper, local pattern variables are not mentioned.

#### Example (related to our demo):
```
symbolic parameters: (box : Box, partition: Partition)
input adornment:     BF
pattern:             box -partitions-> localPartition : Partition -next->partition : Partition
```
In this example, ```partition``` may only be reached via ```localPartition```, which is a local variable and not reflected in the input adornment.
Only 4 BDD variables would be generated in the old reachability analysis (2 for ```box``` and 2 for ```partition```).

**Comment by gervarro** [comment]
> Should we consider the local variables of a pattern as well when analyzing the reachability of a pattern?

Obviously yes, otherwise the algorithm does not make any sense.

> Only 4 BDD variables would be generated in the old reachability analysis

Are you sure that the old reachability analysis ignores local variables? (I have not checked the code.)

**Comment by RolandKluge** [comment]
> Obviously yes, otherwise the algorithm does not make any sense.
OK, then I have been lucky during the test procedure until now.
> Are you sure that the old reachability analysis ignores local variables? (I have not checked the code.)
The old reachability analysis supports *local operations* but, given only the variables in the adornment of the pattern invocation are considered, e.g., when creating the set of BDD variables, the analysis does not support any local variables, yet.

As I remarked in #207, the old reachability analysis was not able to support patterns that contain operations having a different adornment than the pattern invocation.
I have already fixed this.
